apps/antalmanac/src/stores/Schedules.ts

```diff
@@ -461,20 +461,23 @@ export class Schedules {
 
             // Get the course info for each course
             const courseInfoDict = new Map<string, { [sectionCode: string]: CourseInfo }>();
-            for (const [term, courseSet] of Object.entries(courseDict)) {
+
+            const websocRequests = Object.entries(courseDict).map(async ([term, courseSet]) => {
                 const sectionCodes = Array.from(courseSet);
-                // Code from ImportStudyList
                 const courseInfo = getCourseInfo(
                     await queryWebsoc({
                         term: term,
                         sectionCodes: sectionCodes.join(','),
                     })
                 );
```

Comment:
Since both `term` and `sectionCodes` are calculated correctly prior to this function call, it makes the provided object argument cleaner.

```suggestion
                const courseInfo = getCourseInfo(await queryWebsoc({ term, sectionCodes }));
```